### PR TITLE
[ENH] Sort out clustering base class

### LINKED ISSUE #1530: [ENH] `BaseClusterer` inconsistencies with other base classes
There are a few issues present in the `BaseClusterer` class IMO:

- The `predict` function has a `y` input, this is inconsistent with other collection classes and with `predict_proba` from the same class
- The `score` function is functionally different to other bases and is a required method. Can all algorithms produce scores like this? What about pipelines etc.
- `fit_predict` does not have a private equivalent like other classes. Also, by default it seems like this should return `labels_` rather than calling `_predict`?
- `n_clusters` is a required attribute, but not all algorithms have a set number of clusters.

Agreed at the dev meeting 25/10/25:

- [x] The predict function has a y input, this is inconsistent with other collection classes and with predict_proba from the same class
- [ ] The score function is functionally different to other bases and is a required method. Can all algorithms produce scores like this? What about pipelines etc.
- [ ] fit_predict does not have a private equivalent like other classes. Also, by default it seems like this should return labels_ rather than calling _predict?
- [ ] n_clusters is a required attribute, but not all algorithms have a set number of clusters.

### PR DESCRIPTION
**Pull request by TonyBagnall**
Fixes #1530

- [x] The predict function has a y input, this is inconsistent with other collection classes and with predict_proba from the same class
- [x] The score function is functionally different to other bases and is a required method. Can all algorithms produce scores like this? What about pipelines etc.
- [x] fit_predict does not have a private equivalent like other classes. Also, by default it seems like this should return labels_ rather than calling _predict?
- [x] n_clusters is a required attribute, but not all algorithms have a set number of clusters.

### CODE REVIEW

**Review by MatthewMiddlehurst — DISMISSED** [review]
@hadifawaz1999 is this the one you wanted delayed?

Approving for if thats fine. Left some comments.

File: aeon/clustering/deep_learning/_ae_fcn.py [hunk 1]
```diff
@@ -173,10 +173,10 @@ def __init__(
         self.save_last_model = save_last_model
         self.best_file_name = best_file_name
         self.random_state = random_state
+        self.n_clusters = n_clusters
```
[the remark this answers is not in the document]
Comment (hadifawaz1999) [reply aeon/clustering/deep_learning/_ae_fcn.py:176]: i think it should not be added as we're gonna remove it in a few days

**Comment by hadifawaz1999** [comment]
> @hadifawaz1999 is this the one you wanted delayed?
> 
> Approving for if thats fine. Left some comments.

no the one am delaying is Chris's pr of deprecation, but the additions here to deep clustering i think should not be made @TonyBagnall as they're gonna be removed with the purging in a few days once all Aadya's prs are in

**Comment by TonyBagnall** [comment]
could also do this at the same time @chrisholder ?
https://github.com/aeon-toolkit/aeon/issues/2252

File: aeon/clustering/deep_learning/_ae_fcn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisholder) [reply aeon/clustering/deep_learning/_ae_fcn.py:176]: Removing all mention of n_clusters in the dnn clusterers

**Review by hadifawaz1999 — DISMISSED** [review]
LGTM

File: aeon/testing/estimator_checking/_yield_clustering_checks.py [hunk 2]
```diff
@@ -94,6 +99,35 @@ def check_clustering_random_state_deep_learning(estimator, datatype):
...
+    data, and that predict_proba probability estimates add up to one.
+    """
+    estimator = _clone_estimator(estimator)
+
+    # run fit and predict
+    estimator.fit(
+        FULL_TEST_DATA_DICT[datatype]["train"][0],
+        FULL_TEST_DATA_DICT[datatype]["train"][1],
+    )
+    assert hasattr(estimator, "labels_")
```
[the remark this answers is not in the document]
Comment (chrisholder) [reply aeon/testing/estimator_checking/_yield_clustering_checks.py:113]: I have added in additional tests to check fit_predict and predict are equivalent. We may have some issues down the line for clusterers that dont support predict but we can worry about that then...

**Review by hadifawaz1999 — DISMISSED** [review]

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**